sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -329,35 +313,51 @@ def _buildprovider(
     connection_string: Optional[str], endpoint: Optional[str], credential: Optional["TokenCredential"], **kwargs
 ) -> "AzureAppConfigurationProvider":
     # pylint:disable=protected-access
-    provider = AzureAppConfigurationProvider(**kwargs)
+    if connection_string:
+        endpoint = connection_string.split(";")[0].split("=")[1]
+    provider = AzureAppConfigurationProvider(endpoint, **kwargs)
     retry_total = kwargs.pop("retry_total", 2)
     retry_backoff_max = kwargs.pop("retry_backoff_max", 60)
+    replica_discovery_enabled = kwargs.pop("replica_discovery_enabled", True)
 
     if "user_agent" in kwargs:
         user_agent = kwargs.pop("user_agent") + " " + USER_AGENT
     else:
         user_agent = USER_AGENT
 
-    if connection_string:
-        provider._client = AzureAppConfigurationClient.from_connection_string(
-            connection_string,
-            user_agent=user_agent,
-            retry_total=retry_total,
-            retry_backoff_max=retry_backoff_max,
-            **kwargs
+    clients = []
+    if connection_string and endpoint:
+        clients.append(
+            ConfigurationClientWrapper.from_connection_string(
+                endpoint, connection_string, user_agent, retry_total, retry_backoff_max, **kwargs
+            )
         )
+        failover_endpoints = find_auto_failover_endpoints(endpoint, replica_discovery_enabled)
```

Comment:
We are getting the list of failover clients only during startup. But this list needs to be refreshed frequently to make sure any new replicas added or deleted are being accounted for. See the usage of `DiscoverFallbackClients()` in .NET provider for reference: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/0c9e17a9e04b7be98388db7c7dc6e0dffc37939e/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs#L215

Reply:
That and async will be their own PRs I was trying to limit this change.